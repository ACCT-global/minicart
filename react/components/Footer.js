import React, { Component } from 'react'
import { injectIntl, intlShape } from 'react-intl'

class Footer extends Component {
  static propTypes = {
    intl: intlShape.isRequired,
  }

  translate = id => this.props.intl.formatMessage({ id: `dreamstore.${id}` })

  render() {
    const year = new Date().getFullYear()
    return (
      <footer className="w-100 bottom-0 pa7 pa9-l bg-near-white flex items-center">
        <div className="mw8 center">
          <div className="cf tc fw7 flex items-center justify-center">
            {this.translate('poweredBy')}
          </div>
          <div className="tc">{year}</div>
        </div>
      </footer>
    )
  }
}

export default injectIntl(Footer)
