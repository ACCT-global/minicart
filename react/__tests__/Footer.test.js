import React from 'react'
import { mountWithIntl, loadTranslation, setLocale } from 'enzyme-react-intl'
import Footer from '../components/Footer'

describe('Footer test', () => {
  let wrapperPT

  beforeEach(() => {
    loadTranslation('../locales/pt-BR.json')
    setLocale('pt-BR')
    wrapperPT = mountWithIntl(<Footer />)
  })

  it('should render correctly pt-BR', () => {
    expect(wrapperPT.getElement()).toMatchSnapshot()
  })

  it('should be a footer element', () => {
    expect(wrapperPT.find('footer').length).toBe(1)
  })

  it('should render correctly en-US', () => {
    loadTranslation('../locales/en-US.json')
    setLocale('en-US')
    const wrapperEN = mountWithIntl(<Footer />)
    expect(wrapperEN).toMatchSnapshot()
  })

  it('should render correctly es-AR', () => {
    loadTranslation('../locales/es-AR.json')
    setLocale('es-AR')
    const wrapperES = mountWithIntl(<Footer />)
    expect(wrapperES).toMatchSnapshot()
  })
})
