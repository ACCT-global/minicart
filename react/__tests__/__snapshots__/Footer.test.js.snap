// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer test should render correctly en-US 1`] = `
<InjectIntl(Footer)
  intl={
    Object {
      "defaultFormats": Object {},
      "defaultLocale": "en",
      "formatDate": [Function],
      "formatHTMLMessage": [Function],
      "formatMessage": [Function],
      "formatNumber": [Function],
      "formatPlural": [Function],
      "formatRelative": [Function],
      "formatTime": [Function],
      "formats": Object {},
      "formatters": Object {
        "getDateTimeFormat": [Function],
        "getMessageFormat": [Function],
        "getNumberFormat": [Function],
        "getPluralFormat": [Function],
        "getRelativeFormat": [Function],
      },
      "locale": "en-US",
      "messages": Object {
        "dreamstore.buy": "Buy",
        "dreamstore.cart": "Cart",
        "dreamstore.poweredBy": "POWERED BY VTEX",
        "dreamstore.search": "Search",
        "dreamstore.search-placeholder": "Search for products, brands...",
        "dreamstore.soon": "Soon...",
      },
      "now": [Function],
      "textComponent": "span",
    }
  }
>
  <Footer
    intl={
      Object {
        "defaultFormats": Object {},
        "defaultLocale": "en",
        "formatDate": [Function],
        "formatHTMLMessage": [Function],
        "formatMessage": [Function],
        "formatNumber": [Function],
        "formatPlural": [Function],
        "formatRelative": [Function],
        "formatTime": [Function],
        "formats": Object {},
        "formatters": Object {
          "getDateTimeFormat": [Function],
          "getMessageFormat": [Function],
          "getNumberFormat": [Function],
          "getPluralFormat": [Function],
          "getRelativeFormat": [Function],
        },
        "locale": "en-US",
        "messages": Object {
          "dreamstore.buy": "Buy",
          "dreamstore.cart": "Cart",
          "dreamstore.poweredBy": "POWERED BY VTEX",
          "dreamstore.search": "Search",
          "dreamstore.search-placeholder": "Search for products, brands...",
          "dreamstore.soon": "Soon...",
        },
        "now": [Function],
        "textComponent": "span",
      }
    }
  >
    <footer
      className="w-100 bottom-0 pa7 pa9-l bg-near-white flex items-center"
    >
      <div
        className="mw8 center"
      >
        <div
          className="cf tc fw7 flex items-center justify-center"
        >
          POWERED BY VTEX
        </div>
        <div
          className="tc"
        >
          2018
        </div>
      </div>
    </footer>
  </Footer>
</InjectIntl(Footer)>
`;

exports[`Footer test should render correctly es-AR 1`] = `
<InjectIntl(Footer)
  intl={
    Object {
      "defaultFormats": Object {},
      "defaultLocale": "en",
      "formatDate": [Function],
      "formatHTMLMessage": [Function],
      "formatMessage": [Function],
      "formatNumber": [Function],
      "formatPlural": [Function],
      "formatRelative": [Function],
      "formatTime": [Function],
      "formats": Object {},
      "formatters": Object {
        "getDateTimeFormat": [Function],
        "getMessageFormat": [Function],
        "getNumberFormat": [Function],
        "getPluralFormat": [Function],
        "getRelativeFormat": [Function],
      },
      "locale": "es-AR",
      "messages": Object {
        "dreamstore.buy": "Comprar",
        "dreamstore.cart": "Carrito",
        "dreamstore.poweredBy": "IMPULSADO POR VTEX",
        "dreamstore.search": "Buscar",
        "dreamstore.search-placeholder": "Búsqueda por productos, marcas...",
        "dreamstore.soon": "Breve...",
      },
      "now": [Function],
      "textComponent": "span",
    }
  }
>
  <Footer
    intl={
      Object {
        "defaultFormats": Object {},
        "defaultLocale": "en",
        "formatDate": [Function],
        "formatHTMLMessage": [Function],
        "formatMessage": [Function],
        "formatNumber": [Function],
        "formatPlural": [Function],
        "formatRelative": [Function],
        "formatTime": [Function],
        "formats": Object {},
        "formatters": Object {
          "getDateTimeFormat": [Function],
          "getMessageFormat": [Function],
          "getNumberFormat": [Function],
          "getPluralFormat": [Function],
          "getRelativeFormat": [Function],
        },
        "locale": "es-AR",
        "messages": Object {
          "dreamstore.buy": "Comprar",
          "dreamstore.cart": "Carrito",
          "dreamstore.poweredBy": "IMPULSADO POR VTEX",
          "dreamstore.search": "Buscar",
          "dreamstore.search-placeholder": "Búsqueda por productos, marcas...",
          "dreamstore.soon": "Breve...",
        },
        "now": [Function],
        "textComponent": "span",
      }
    }
  >
    <footer
      className="w-100 bottom-0 pa7 pa9-l bg-near-white flex items-center"
    >
      <div
        className="mw8 center"
      >
        <div
          className="cf tc fw7 flex items-center justify-center"
        >
          IMPULSADO POR VTEX
        </div>
        <div
          className="tc"
        >
          2018
        </div>
      </div>
    </footer>
  </Footer>
</InjectIntl(Footer)>
`;

exports[`Footer test should render correctly pt-BR 1`] = `
<InjectIntl(Footer)
  intl={
    Object {
      "defaultFormats": Object {},
      "defaultLocale": "en",
      "formatDate": [Function],
      "formatHTMLMessage": [Function],
      "formatMessage": [Function],
      "formatNumber": [Function],
      "formatPlural": [Function],
      "formatRelative": [Function],
      "formatTime": [Function],
      "formats": Object {},
      "formatters": Object {
        "getDateTimeFormat": [Function],
        "getMessageFormat": [Function],
        "getNumberFormat": [Function],
        "getPluralFormat": [Function],
        "getRelativeFormat": [Function],
      },
      "locale": "pt-BR",
      "messages": Object {
        "dreamstore.buy": "Comprar",
        "dreamstore.cart": "Carrinho",
        "dreamstore.poweredBy": "FORNECIDO POR VTEX",
        "dreamstore.search": "Buscar",
        "dreamstore.search-placeholder": "Busque por produtos, marcas...",
        "dreamstore.soon": "Em breve...",
      },
      "now": [Function],
      "textComponent": "span",
    }
  }
/>
`;
